gemfirexd/core/src/main/java/com/pivotal/gemfirexd/internal/impl/sql/GenericStatement.java

```diff
@@ -151,7 +151,7 @@ Licensed to the Apache Software Foundation (ASF) under one or more
             Pattern.compile(".*PUT\\s+INTO\\s+(TABLE)?.*SELECT\\s+.*",
                 Pattern.CASE_INSENSITIVE);
         private static final Pattern EXECUTION_ENGINE_STORE_HINT =
-            Pattern.compile(".*EXECUTIONENGINE(\\s+)?+=(\\s+)?+STORE\\s+.*",
+            Pattern.compile(".*EXECUTIONENGINE(\\s+)?+=(\\s+)?+STORE.*[\\r\\n].*",
```

Comment:
The one suggested by you.
if the string contains newline/returns then .* doesn't match due to '\n'.
Figured out that ".*\\bEXECUTIONENGINE(\\s+)?+=(\\s+)?+STORE\\s*\\b.*[\r\n]?.*" is matching.